### PR TITLE
Various ctransformers fixes

### PR DESCRIPTION
**Pull request by netrunnereve**
I had to look into the code to get ctransformers to work, and ended up fixing a few things as well. Truthfully though when GGUF ends up getting merged llama.cpp should automatically support MPT and Falcon out of the box.

- added proper documentation
- added ctransformers models to `config.yaml` (note that I didn't try all of them :woman_shrugging:)
- set context length for ctransformers

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -235,22 +235,33 @@ Optionally, you can use the following command-line flags:
...
-#### llama.cpp
+#### GGML (for llama.cpp and ctransformers)
 
 | Flag        | Description |
 |-------------|-------------|
 | `--threads` | Number of threads to use. |
 | `--n_batch` | Maximum number of prompt tokens to batch together when calling llama_eval. |
+| `--n-gpu-layers N_GPU_LAYERS` | Number of layers to offload to the GPU. Only works if llama-cpp-python was compiled with BLAS. Set this to 1000000000 to offload all layers to the GPU. |
+| `--n_ctx N_CTX` | Size of the prompt context. |
+| `--llama_cpp_seed SEED` | Seed for llama-cpp and ctransformers models. Default 0 (random). |
```
Comment (oobabooga) [inline README.md:246]: llama_cpp_seed is not used by ctransformers.

**Comment by oobabooga** [comment]
It looks good now, thank you!

**Comment by cal066** [comment]
@oobabooga  @netrunnereve 
I just noticed it is not gpt_bigcode but gptbigcode, see the [LLM class](https://github.com/marella/ctransformers/blob/aa38adef26acace4c4343cf6648ef153e091a290/models/llm.cc#L28). The models/config.yml probably needs updating.

gptbigcode and starcoder is equivalent, but `modules/loaders.py` still references starcoder.

**Comment by cal066** [comment]
Looks like its more of a cosmetic issue since ctransformers strip non-alphanumeric characters for model_type comparison, I have created #3567 to resolve the consistency.

**Pull request merged**